packages/core/lib/commands/debug/run.js

```diff
@@ -1,14 +1,19 @@
 module.exports = async function (options) {
   const { promisify } = require("util");
   const debugModule = require("debug");
+  const mergeConfigNetwork = require("./mergeConfigNetwork");
+  const loadConfig = require("./loadConfig");
   const debug = debugModule("lib:commands:debug");
 
   const { Environment } = require("@truffle/environment");
-  const Config = require("@truffle/config");
-
   const { CLIDebugger } = require("../../debug");
 
-  const config = Config.detect(options);
+  const config = loadConfig(options);
```

Comment:
I still think there should be something prior to this that checks that you didn't set both `--network` and `--url`.

Reply:
added a check. Not very happy with the copy/wording though. Let me know if you have a better idea :)